### PR TITLE
test: add benchmark

### PR DESCRIPTION
**Pull request by ctavan**
Add a simple benchmark to be able to catch order-of-magnitude
performance regressions.

### CODE REVIEW

**Comment by ctavan** [comment]
Agree @awwit. I think if at all, this benchmark will only allow us to compare before/after of a certain change on the same machine.

I don't think we can get comparable results over time and I also don't intend to do this.

File: README_js.md [hunk 1]
```diff
@@ -353,6 +353,17 @@ import { v4 as uuidv4 } from 'uuid';
 Workers](https://caniuse.com/#feat=cryptography) and we are not aware of a polyfill (let us know if
 you find one, please).
 
+## Developing
```
Comment (broofa) [inline README_js.md:356]: This isn't of interest to most readers.  Move to [PR template](https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository) (preferably the hidden `.github/pull_request_template.md` file).

File: examples/benchmark/benchmark.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/* global Benchmark:false, uuidv1:false, uuidv3:false, uuidv4:false, uuidv5:false */
+const Benchmark = (typeof window !== 'undefined' && window.Benchmark) || require('benchmark');
+const uuidv1 = (typeof window !== 'undefined' && window.uuidv1) || require('uuid').v1;
+const uuidv4 = (typeof window !== 'undefined' && window.uuidv4) || require('uuid').v4;
+const uuidv3 = (typeof window !== 'undefined' && window.uuidv3) || require('uuid').v3;
+const uuidv5 = (typeof window !== 'undefined' && window.uuidv5) || require('uuid').v5;
+
+const MY_NAMESPACE = '55238d15-c926-4598-b49d-cf4e913ba13c';
```
Comment (broofa) [inline examples/benchmark/benchmark.js:8]: No need to performance test DNS, URL, and custom namespaces.  The codepath will be the same regardless of the namespace id, so just pick one (e.g. DNS) and test that.

```suggestion
```

File: examples/benchmark/benchmark.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    // ... using predefined URL namespace (for, well, URLs)
+    uuidv3('http://example.com/hello', uuidv3.URL);
+  })
+  .add('uuidv3() MY_NAMESPACE', function () {
+    // ... using a custom namespace
+    //
+    // Note: Custom namespaces should be a UUID string specific to your application!
+    // E.g. the one here was generated using this modules `uuid` CLI.
+    uuidv3('Hello, World!', MY_NAMESPACE);
+  })
```
Comment (broofa) [inline examples/benchmark/benchmark.js:39]:
```suggestion
```

File: examples/benchmark/benchmark.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    uuidv5('http://example.com/hello', uuidv5.URL);
+  })
+  .add('uuidv5() MY_NAMESPACE', function () {
+    // ... using a custom namespace
+    //
+    // Note: Custom namespaces should be a UUID string specific to your application!
+    // E.g. the one here was generated using this modules `uuid` CLI.
+    // const MY_NAMESPACE = '1b671a64-40d5-491e-99b0-da01ff1f3341';
+    uuidv5('Hello, World!', MY_NAMESPACE);
+  })
```
Comment (broofa) [inline examples/benchmark/benchmark.js:55]:
```suggestion
```

**Comment by broofa** [comment]
FWIW, here's the results for node 12.10.0 on my laptop (same laptop I used for the [other results I posted](https://github.com/uuidjs/uuid/pull/427#issuecomment-621572570))
```
uuidv1() x 1,472,274 ops/sec ±0.62% (91 runs sampled)
uuidv1() fill existing array x 4,389,392 ops/sec ±0.44% (87 runs sampled)
uuidv4() x 348,408 ops/sec ±0.75% (87 runs sampled)
uuidv4() fill existing array x 390,717 ops/sec ±1.32% (84 runs sampled)
uuidv3() DNS x 137,150 ops/sec ±0.49% (88 runs sampled)
uuidv3() URL x 126,044 ops/sec ±0.75% (89 runs sampled)
uuidv3() MY_NAMESPACE x 129,816 ops/sec ±0.45% (92 runs sampled)
uuidv5() DNS x 135,630 ops/sec ±0.68% (88 runs sampled)
uuidv5() URL x 123,146 ops/sec ±1.03% (90 runs sampled)
uuidv5() MY_NAMESPACE x 127,863 ops/sec ±0.67% (84 runs sampled)
Fastest is uuidv1() fill existing array
```

File: examples/benchmark/benchmark.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
+/* global Benchmark:false, uuidv1:false, uuidv3:false, uuidv4:false, uuidv5:false */
+const Benchmark = (typeof window !== 'undefined' && window.Benchmark) || require('benchmark');
+const uuidv1 = (typeof window !== 'undefined' && window.uuidv1) || require('uuid').v1;
+const uuidv4 = (typeof window !== 'undefined' && window.uuidv4) || require('uuid').v4;
+const uuidv3 = (typeof window !== 'undefined' && window.uuidv3) || require('uuid').v3;
+const uuidv5 = (typeof window !== 'undefined' && window.uuidv5) || require('uuid').v5;
+
+const MY_NAMESPACE = '55238d15-c926-4598-b49d-cf4e913ba13c';
+
```
Comment (broofa) [inline examples/benchmark/benchmark.js:9]: It's a little unclear that anything is happening at first.

```suggestion
console.log('Starting.  Tests take ~1 minute to run ...');

```

**Pull request merged**